### PR TITLE
API Simplification

### LINKED ISSUE #301: API Clarifications
I think the names we have chosen for all these `node_neighbors` and `space_neighgbhors` are not optimal and can be confusing. I'd suggest the alternatives:

* `neighborhing_agents`
* `neighborhing_positions`

or even better, to save space and not having to type one of the most difficult words in the english lexicon:

* `nearby_agents`
* `nearby_positions`

which can deprecate the existing names.

### PR DESCRIPTION
**Pull request by Libbum**
Closes #301 (when complete).

Note that `@deprecate` is silent from julia 1.5 and up, unless you run `julia --depwarn=yes`

- [x] space_neighbors -> nearby_agents
- [x] node_neighbors -> nearby_positions
- [x] get_node_contents -> agents_in_pos
- [x] Multi agent model -> mixed agent model (if it is somewhere) [Only in the old paper and old values in the CHANGELOG. Didn't touch them]
- [x] pick_empty -> random_empty
- [x] find_empty_nodes -> empty_nodes [**empty_positions** yeah?]
- [x] Replace all instances of the word "node" with the word "position" (and also nodes with positions). This will change both wording in documentation string but also function names, e.g. nodes -> positions. [Growing bacteria uses the word 'node' in a different context, so has been left in; SIR is speaking directly about graph plots, node stays when it context]
- [x] Check for *vertex*, *cell* and other graph/grid specific nomenclature and make it space agnostic [Sugarscape needs work. It's strongly coupled to the old ways. Ali, can you take a look at it?]
- [x] Complete read-through of documentation & examples to validate changes.
    - [x] Schelling
    - [x] Sugarscape
    - [x] SIR
    - [x] Social Distance
    - [x] Wealth Distribution
    - [x] Forest Fire
    - [x] Game of Life
    - [x] Wright-Fisher
    - [x] HK
    - [x] Flocking
    - [x] Daisyworld
    - [x] Wolf Sheep
    - [x] Bacteria Growth
    - [x] Opinion Spread
- [x] Verify tests still make sense
- [x] Make sure models run

### CODE REVIEW

**Comment by Libbum** [comment]
I'm leaving this for tonight now.

Once the docs build, we'll need to verify the models are running correctly (they build and step at least), and all the docs now make sense with all of these changes.

@kavir1698, can you take a look at sugarscape though? It was too tied to node/index positions rather than our (x,y) protocol now&mdash;I wasn't sure how to get it functioning.

Other than verification, I think this is 'done'. Just small omissions to pick up is my guess.

**Comment by Libbum** [comment]
`plotabm` is one of the five places I've identified that uses `get_node_agents` rather than `get_node_contents`.

This is making our plots fail in many instances in the documentation. 
If I add it back, what should we call it? get_node_contents -> agents_in_pos has already sort of shot us in the foot for names separating the ids and the actual agents.

Alternatively, I could do a PR on AgentsPlots with updated function definitions, and track that branch in the documentation builder here, until things settle.
 
Thoughts?

**Comment by Datseris** [comment]
An alternative is to rename `nearby_agents, agents_in_pos` to have `ids` instead of `agents`, and define the functions with agents to return `(model[id] for id in ...)` instead.

**Comment by Libbum** [comment]
Could do that. So that's a vote to adding back a `get_node_agents` function back in?

**Comment by Datseris** [comment]
Yes, although it will be called `agents_in_position`.

**Comment by Libbum** [comment]
Here's the changes I'll do then, sound good?

space_neighbors -> nearby_agents -> nearby_ids
get_node_contents -> agents_in_pos -> ids_in_position
get_node_agents -> agents_in_position

**Comment by Datseris** [comment]
yeap, sounds perfect to me, you could also then define the simple `nearby_agents` as 
```
nearby_agents(a, model) = (model[id] for id in nearby_ids(a, model))
```

**Comment by Libbum** [comment]
New changes add `agents_in_position`, Closes #295.

**Comment by Libbum** [comment]
Will close #294.

**Comment by Libbum** [comment]
```julia
for dia in 1:(agent.vision)
   nn = node_neighbors(posvertex, model, dia)
   ...
end
```
@kavir1698 what was the purpose of this loop in sugarscape? Is it meant to loop over the directions? Or is it meant to look further afield? Like expanding concentric circles?

**Comment by Libbum** [comment]
Unless you guys can find an issue, I think this is ready now.

There are some spots in the documentation that need updating for 4.0, but that's not part of this PR I think. We can track that in #318.

File: src/spaces/discrete.jl [hunk 1]
```diff
@@ -33,6 +33,15 @@ function positions(model::ABM{<:AbstractAgent,<:DiscreteSpace}, by::Symbol)
...
 end
 
+"""
+    ids_in_position(position, model::ABM{A, <:DiscreteSpace})
+    ids_in_position(agent, model::ABM{A, <:DiscreteSpace})
+
+Return the ids of agents in the position corresponding to `position` or position
+of `agent`.
+"""
+ids_in_position(agent::A, model) where {A<:AbstractAgent} = ids_in_position(agent.pos, model)
```
Comment (Datseris) [inline src/spaces/discrete.jl:43]: Please separate the docstring of `ids_in_position(agent, ...)` to simply state:
```
ids_in_position(agent::AbstractAgent, model) = ids_in_position(agent.pos, model)
```

**Comment by Libbum** [comment]
I've found a few omissions based on the documentation run, fixing up those now.

**Comment by Libbum** [comment]
I somehow cancelled your review George. Can you tell me again what you wanted to do with the docstring?

**Comment by Datseris** [comment]
> I somehow cancelled your review George. Can you tell me again what you wanted to do with the docstring?

Nah I was wrong so I deleted it :D

**Comment by Datseris** [comment]
What were the changes you did to `plotabm`? Can you PR them over at AgentsPlots.jl?

**Comment by Libbum** [comment]
> What were the changes you did to plotabm?

Nothing :) Since we added in the `@deprecate get_node_agents agents_in_position` clause, everything works without needing to do anything for now.

**Comment by Datseris** [comment]
Everything seems fine online: https://juliadynamics.github.io/Agents.jl/previews/PR317/api/

**Comment by Datseris** [comment]
Sugarscape can be taken care of in a new PR, better to have name changes in before more PRs happen.

**Comment by Libbum** [comment]
Sugarscape is done anyway. Would be good for Ali to verify, but I think I fixed it.

**Pull request merged**